doc/contributing.rst

```diff
@@ -1,221 +1,808 @@
+should use ``snake_case`` (``n_samples`` rather than ``nsamples``). Avoid
+single-character variable names, unless inside a :term:`comprehension <list
+comprehension>` or :ref:`generator <tut-generators>`.
+
+
+Follow NumPy style for docstrings
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In most cases imitating existing docstrings will be sufficient, but consult the
+`Numpy docstring style guidelines`_ for more complicated formatting such as
+embedding example code, citing references, or including rendered mathematics.
+Private function/method docstrings may be brief for simple functions/methods,
+but complete docstrings are appropriate when private functions/methods are
+relatively complex. To run some basic tests on documentation, you can use:
+
+.. code-block:: console
+
+    $ pytest mne/tests/test_docstring_parameters
+    $ make docstyle
+
+
+Cross-reference everywhere
+~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-If you are working on examples or tutorials, you can build specific examples
-with:
+Both the docstrings and dedicated documentation pages (tutorials, how-to
+examples, discussions, and glossary) should include cross-references to any
+mentioned module, class, function, method, attribute, or documentation page.
+There are sphinx directives for all of these (``:mod:``, ``:class:``,
+``:func:``, ``:meth:``, ``:attr:``, ``:doc:``) as well as a generic
+cross-reference directive (``:ref:``) for linking to specific sections of a
+documentation page. MNE-Python also uses Intersphinx_, so you can (and should)
+cross-reference to Python built-in classes and functions as well as API
+elements in :mod:`NumPy <numpy>`, :mod:`SciPy <scipy>`, etc. See the Sphinx
+configuration file (:file:`doc/conf.py`) for the list of Intersphinx projects
+we link to.
+
+.. warning::
+
+    Some API elements have multiple exposure points (for example,
+    ``mne.set_config`` and ``mne.utils.set_config``). For cross-references to
+    work, they must match an entry in :file:`doc/python_reference.rst` (thus
+    ``:func:`mne.set_config``` will work but ``:func:`mne.utils.set_config```
+    will not).
+
+
+Other style guidance
+~~~~~~~~~~~~~~~~~~~~
+
+- Use single quotes whenever possible.
+
+- Prefer :ref:`generators <tut-generators>` or
+  :term:`comprehensions <list comprehension>` over :func:`filter`, :func:`map`
+  and other functional idioms.
+
+- Use explicit functional constructors for builtin containers to improve
+  readability (e.g., :ref:`list() <func-list>`, :ref:`dict() <func-dict>`,
+  :ref:`set() <func-set>`).
+
+- Avoid nested functions or class methods if possible — use private functions
+  instead.
+
+- Avoid ``*args`` and ``**kwargs`` in function/method signatures.
+
+
+Code organization
+^^^^^^^^^^^^^^^^^
+
+Importing
+~~~~~~~~~
+
+Import modules in this order:
+
+1. Python built-in (``os``, ``copy``, ``functools``, etc)
+2. standard scientific (``numpy as np``, ``scipy.signal``, etc)
+3. others
+4. MNE-Python imports (e.g., ``from .pick import pick_types``)
```

Comment:
I would add that relative imports should be replaced with full imports in test files.